JSON deserialization error on pristine Graylog beta.3-SNAPSHOT installation

## Expected Behavior

Server startup with a pristine installation should work without errors.

## Current Behavior

When starting the server I see the following JSON deserialization errors with `IndexSetCreated` events.

```
2016-12-19 14:07:15,783 INFO : org.graylog2.migrations.V20161116172200_CreateDefaultStreamMigration - Successfully created default stream: All messages
2016-12-19 14:07:15,840 INFO : org.graylog2.migrations.V20161124104700_AddRetentionRotationAndDefaultFlagToIndexSetMigration - Adding rotation_strategy_class <org.graylog2.indexer.rotation.strategies.MessageCountRotationStrategy> to index set <5857db8316049d4995cd1ff4>
2016-12-19 14:07:15,840 INFO : org.graylog2.migrations.V20161124104700_AddRetentionRotationAndDefaultFlagToIndexSetMigration - Adding retention_strategy_class <org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy> to index set <5857db8316049d4995cd1ff4>
2016-12-19 14:07:15,855 INFO : org.graylog2.migrations.V20161124104700_AddRetentionRotationAndDefaultFlagToIndexSetMigration - Setting index set <5857db8316049d4995cd1ff4> as default
2016-12-19 14:07:15,916 INFO : org.graylog2.migrations.V20161125142400_EmailAlarmCallbackMigration - No streams needed to be migrated.
2016-12-19 14:07:15,952 INFO : org.graylog2.migrations.V20161125161400_AlertReceiversMigration - No streams needed to be migrated.
2016-12-19 14:07:15,995 INFO : org.graylog2.migrations.V20161130141500_DefaultStreamRecalcIndexRanges - Cluster not connected yet, delaying migration until it is reachable.
2016-12-19 14:07:16,565 ERROR: org.graylog2.events.ClusterEventPeriodical - Error while deserializing payload
java.lang.IllegalArgumentException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3589) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3508) ~[graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.extractPayload(ClusterEventPeriodical.java:207) [graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.doRun(ClusterEventPeriodical.java:158) [graylog.jar:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:74) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1410) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.JSR310DeserializerBase._rethrowDateTimeException(JSR310DeserializerBase.java:79) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:208) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
Caused by: java.time.format.DateTimeParseException: Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949) ~[?:1.8.0_111]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1777) ~[?:1.8.0_111]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:202) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
2016-12-19 14:07:16,568 WARN : org.graylog2.events.ClusterEventPeriodical - Couldn't extract payload of cluster event with ID <5857db8316049d4995cd1ff5>
2016-12-19 14:07:16,568 WARN : org.graylog2.events.ClusterEventPeriodical - Invalid payload in cluster event: ClusterEvent{id=5857db8316049d4995cd1ff5, timestamp=1482152835728, producer=28164cbe-4ad9-4c9c-a76e-088655aa7889, consumers=[], eventClass=org.graylog2.indexer.indexset.events.IndexSetCreatedEvent, payload={index_set={_id=5857db8316049d4995cd1ff4, title=Default index set, description=The Graylog default index set, index_prefix=xgraylog, shards=4, replicas=0, rotation_strategy={type=org.graylog2.indexer.rotation.strategies.MessageCountRotationStrategyConfig, max_docs_per_index=20000000}, retention_strategy={type=org.graylog2.indexer.retention.strategies.DeletionRetentionStrategyConfig, max_number_of_indices=20}, creation_date=Mon Dec 19 14:07:15 CET 2016, writable=true}}}
2016-12-19 14:07:16,716 ERROR: org.graylog2.events.ClusterEventPeriodical - Error while deserializing payload
java.lang.IllegalArgumentException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3589) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3508) ~[graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.extractPayload(ClusterEventPeriodical.java:207) [graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.doRun(ClusterEventPeriodical.java:158) [graylog.jar:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:74) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1410) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.JSR310DeserializerBase._rethrowDateTimeException(JSR310DeserializerBase.java:79) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:208) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
Caused by: java.time.format.DateTimeParseException: Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949) ~[?:1.8.0_111]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1777) ~[?:1.8.0_111]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:202) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
2016-12-19 14:07:16,717 WARN : org.graylog2.events.ClusterEventPeriodical - Couldn't extract payload of cluster event with ID <5857db8316049d4995cd1ff9>
2016-12-19 14:07:16,717 WARN : org.graylog2.events.ClusterEventPeriodical - Invalid payload in cluster event: ClusterEvent{id=5857db8316049d4995cd1ff9, timestamp=1482152835872, producer=28164cbe-4ad9-4c9c-a76e-088655aa7889, consumers=[], eventClass=org.graylog2.indexer.indexset.events.IndexSetCreatedEvent, payload={index_set={_id=5857db8316049d4995cd1ff4, title=Default index set, description=The Graylog default index set, index_prefix=xgraylog, shards=4, replicas=0, rotation_strategy={type=org.graylog2.indexer.rotation.strategies.MessageCountRotationStrategyConfig, max_docs_per_index=20000000}, retention_strategy={type=org.graylog2.indexer.retention.strategies.DeletionRetentionStrategyConfig, max_number_of_indices=20}, creation_date=Mon Dec 19 14:07:15 CET 2016, writable=true}}}
2016-12-19 14:07:16,794 INFO : org.mongodb.driver.connection - Opened connection [connectionId{localValue:7, serverValue:194}] to 127.0.0.1:27017
2016-12-19 14:07:16,831 INFO : org.mongodb.driver.connection - Opened connection [connectionId{localValue:8, serverValue:195}] to 127.0.0.1:27017
2016-12-19 14:07:16,859 ERROR: org.graylog2.events.ClusterEventPeriodical - Error while deserializing payload
java.lang.IllegalArgumentException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3589) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:3508) ~[graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.extractPayload(ClusterEventPeriodical.java:207) [graylog.jar:?]
	at org.graylog2.events.ClusterEventPeriodical.doRun(ClusterEventPeriodical.java:158) [graylog.jar:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_111]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_111]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not deserialize value of type java.time.ZonedDateTime from String "2016-12-19T13:07:15.697+0000": Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.graylog2.indexer.indexset.events.IndexSetCreatedEvent["index_set"]->org.graylog2.indexer.indexset.IndexSetConfig["creation_date"])
	at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:74) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:1410) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.JSR310DeserializerBase._rethrowDateTimeException(JSR310DeserializerBase.java:79) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:208) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
Caused by: java.time.format.DateTimeParseException: Text '2016-12-19T13:07:15.697+0000' could not be parsed at index 23
	at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1949) ~[?:1.8.0_111]
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1777) ~[?:1.8.0_111]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:202) ~[graylog.jar:?]
	at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:48) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:499) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:511) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:396) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1198) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:314) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:148) ~[graylog.jar:?]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:3584) ~[graylog.jar:?]
	... 11 more
2016-12-19 14:07:16,867 WARN : org.graylog2.events.ClusterEventPeriodical - Couldn't extract payload of cluster event with ID <5857db8416049d4995cd1ffd>
2016-12-19 14:07:16,867 WARN : org.graylog2.events.ClusterEventPeriodical - Invalid payload in cluster event: ClusterEvent{id=5857db8416049d4995cd1ffd, timestamp=1482152836016, producer=28164cbe-4ad9-4c9c-a76e-088655aa7889, consumers=[], eventClass=org.graylog2.indexer.indexset.events.IndexSetCreatedEvent, payload={index_set={_id=5857db8316049d4995cd1ff4, title=Default index set, description=The Graylog default index set, index_prefix=xgraylog, shards=4, replicas=0, rotation_strategy_class=org.graylog2.indexer.rotation.strategies.MessageCountRotationStrategy, rotation_strategy={type=org.graylog2.indexer.rotation.strategies.MessageCountRotationStrategyConfig, max_docs_per_index=20000000}, retention_strategy_class=org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy, retention_strategy={type=org.graylog2.indexer.retention.strategies.DeletionRetentionStrategyConfig, max_number_of_indices=20}, creation_date=Mon Dec 19 14:07:15 CET 2016, writable=true}}}
2016-12-19 14:07:18,435 WARN : org.elasticsearch.discovery - [graylog-28164cbe-4ad9-4c9c-a76e-088655aa7889] waited for 3s and no initial state was set by the discovery
2016-12-19 14:07:18,435 INFO : org.elasticsearch.node - [graylog-28164cbe-4ad9-4c9c-a76e-088655aa7889] started
```

## Steps to Reproduce (for bugs)

1. Enable DEBUG logging for `org.graylog2.events.ClusterEventPeriodical`
2. Start Graylog with an empty database
3. Look for DEBUG messages indicating an error

## Your Environment

* Graylog Version: 2.2.0-beta.3-SNAPSHOT+460cac6

--- a/graylog2-server/src/main/java/org/graylog2/cluster/ClusterConfigServiceImpl.java
+++ b/graylog2-server/src/main/java/org/graylog2/cluster/ClusterConfigServiceImpl.java
@@ -16,12 +16,14 @@
  */
 package org.graylog2.cluster;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableSet;
 import com.google.common.eventbus.EventBus;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.mongodb.DBCollection;
 import com.mongodb.WriteConcern;
+
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.events.ClusterEventBus;
@@ -39,8 +41,9 @@ import org.mongojack.WriteResult;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.inject.Inject;
 import java.util.Set;
+
+import javax.inject.Inject;
 
 import static com.google.common.base.MoreObjects.firstNonNull;
 import static com.google.common.base.Preconditions.checkNotNull;
@@ -61,18 +64,17 @@ public class ClusterConfigServiceImpl implements ClusterConfigService {
     public ClusterConfigServiceImpl(final MongoJackObjectMapperProvider mapperProvider,
                                     final MongoConnection mongoConnection,
                                     final NodeId nodeId,
-                                    final ObjectMapper objectMapper,
                                     final ChainingClassLoader chainingClassLoader,
                                     final ClusterEventBus clusterEventBus) {
         this(JacksonDBCollection.wrap(prepareCollection(mongoConnection), ClusterConfig.class, String.class, mapperProvider.get()),
-                nodeId, objectMapper, chainingClassLoader, clusterEventBus);
+                nodeId, mapperProvider.get(), chainingClassLoader, clusterEventBus);
     }
 
-    ClusterConfigServiceImpl(final JacksonDBCollection<ClusterConfig, String> dbCollection,
-                             final NodeId nodeId,
-                             final ObjectMapper objectMapper,
-                             final ChainingClassLoader chainingClassLoader,
-                             final EventBus clusterEventBus) {
+    private ClusterConfigServiceImpl(final JacksonDBCollection<ClusterConfig, String> dbCollection,
+                                     final NodeId nodeId,
+                                     final ObjectMapper objectMapper,
+                                     final ChainingClassLoader chainingClassLoader,
+                                     final EventBus clusterEventBus) {
         this.nodeId = checkNotNull(nodeId);
         this.dbCollection = checkNotNull(dbCollection);
         this.objectMapper = checkNotNull(objectMapper);

--- a/graylog2-server/src/main/java/org/graylog2/events/ClusterEventPeriodical.java
+++ b/graylog2-server/src/main/java/org/graylog2/events/ClusterEventPeriodical.java
@@ -16,11 +16,12 @@
  */
 package org.graylog2.events;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.eventbus.DeadEvent;
 import com.google.common.eventbus.EventBus;
 import com.google.common.eventbus.Subscribe;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.mongodb.BasicDBList;
 import com.mongodb.BasicDBObject;
 import com.mongodb.DB;
@@ -28,6 +29,7 @@ import com.mongodb.DBCollection;
 import com.mongodb.DBObject;
 import com.mongodb.MongoException;
 import com.mongodb.WriteConcern;
+
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.plugin.periodical.Periodical;
@@ -62,15 +64,14 @@ public class ClusterEventPeriodical extends Periodical {
     public ClusterEventPeriodical(final MongoJackObjectMapperProvider mapperProvider,
                                   final MongoConnection mongoConnection,
                                   final NodeId nodeId,
-                                  final ObjectMapper objectMapper,
                                   final ChainingClassLoader chainingClassLoader,
                                   final EventBus serverEventBus,
                                   final ClusterEventBus clusterEventBus) {
         this(JacksonDBCollection.wrap(prepareCollection(mongoConnection), ClusterEvent.class, String.class, mapperProvider.get()),
-                nodeId, objectMapper, chainingClassLoader, serverEventBus, clusterEventBus);
+                nodeId, mapperProvider.get(), chainingClassLoader, serverEventBus, clusterEventBus);
     }
 
-    ClusterEventPeriodical(final JacksonDBCollection<ClusterEvent, String> dbCollection,
+    private ClusterEventPeriodical(final JacksonDBCollection<ClusterEvent, String> dbCollection,
                            final NodeId nodeId,
                            final ObjectMapper objectMapper,
                            final ChainingClassLoader chainingClassLoader,

--- a/graylog2-server/src/test/java/org/graylog2/cluster/ClusterConfigServiceImplTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/cluster/ClusterConfigServiceImplTest.java
@@ -16,8 +16,9 @@
  */
 package org.graylog2.cluster;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.eventbus.Subscribe;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
@@ -25,6 +26,7 @@ import com.mongodb.BasicDBObjectBuilder;
 import com.mongodb.DBCollection;
 import com.mongodb.DBObject;
 import com.mongodb.WriteConcern;
+
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.database.MongoConnectionRule;
@@ -84,7 +86,6 @@ public class ClusterConfigServiceImplTest {
                 provider,
                 mongoRule.getMongoConnection(),
                 nodeId,
-                objectMapper,
                 new ChainingClassLoader(getClass().getClassLoader()),
                 clusterEventBus
         );

--- a/graylog2-server/src/test/java/org/graylog2/events/ClusterEventPeriodicalTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/events/ClusterEventPeriodicalTest.java
@@ -16,11 +16,12 @@
  */
 package org.graylog2.events;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.eventbus.DeadEvent;
 import com.google.common.eventbus.EventBus;
 import com.google.common.eventbus.Subscribe;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
@@ -28,6 +29,7 @@ import com.mongodb.BasicDBObjectBuilder;
 import com.mongodb.DBCollection;
 import com.mongodb.DBObject;
 import com.mongodb.WriteConcern;
+
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.database.MongoConnection;
 import org.graylog2.database.MongoConnectionRule;
@@ -93,7 +95,6 @@ public class ClusterEventPeriodicalTest {
                 provider,
                 mongoRule.getMongoConnection(),
                 nodeId,
-                objectMapper,
                 new ChainingClassLoader(getClass().getClassLoader()),
                 serverEventBus,
                 clusterEventBus

--- a/graylog2-server/src/test/java/org/graylog2/indexer/indexset/MongoIndexSetServiceTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/indexer/indexset/MongoIndexSetServiceTest.java
@@ -16,11 +16,13 @@
  */
 package org.graylog2.indexer.indexset;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.eventbus.Subscribe;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
+
 import org.bson.types.ObjectId;
 import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
 import org.graylog2.buffers.processors.fakestreams.FakeStream;
@@ -84,7 +86,7 @@ public class MongoIndexSetServiceTest {
     public void setUp() throws Exception {
         clusterEventBus = new ClusterEventBus();
         clusterConfigService = new ClusterConfigServiceImpl(objectMapperProvider, mongoRule.getMongoConnection(),
-                nodeId, objectMapper, new ChainingClassLoader(getClass().getClassLoader()), clusterEventBus);
+                nodeId, new ChainingClassLoader(getClass().getClassLoader()), clusterEventBus);
         indexSetService = new MongoIndexSetService(mongoRule.getMongoConnection(), objectMapperProvider, streamService, clusterConfigService, clusterEventBus);
     }
 

--- a/graylog2-server/src/test/java/org/graylog2/migrations/V20161215163900_MoveIndexSetDefaultConfigTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/migrations/V20161215163900_MoveIndexSetDefaultConfigTest.java
@@ -70,7 +70,7 @@ public class V20161215163900_MoveIndexSetDefaultConfigTest {
     @Before
     public void setUp() throws Exception {
         this.clusterConfigService = spy(new ClusterConfigServiceImpl(objectMapperProvider,
-                fongoRule.getConnection(), nodeId, objectMapper,
+                fongoRule.getConnection(), nodeId,
                 new ChainingClassLoader(getClass().getClassLoader()), new ClusterEventBus()));
 
         this.collection = fongoRule.getDatabase().getCollection("index_sets");

